src/components/button/config/generate.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright IBM Corp. 2018, 2019
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Template configuration generate functions for Button component
+ * @module button/generate
+ */
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+// TODO JSDoc
+export function generate(options) {
+  let opts = merge(
+    {
+      element: 'button',
+      content: 'Button',
+      attributes: {},
+      disabled: false,
+      type: 'button',
```

Comment:
this would be a `parameter` that a `generate` function would receive, and `generate` would convert this param to 

```
attributes: {
  type: 'button'
}
```

so, wouldn't be needed in a `default` config.